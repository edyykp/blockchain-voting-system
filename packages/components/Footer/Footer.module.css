.footer {
  background-color: var(--secondary);
  line-height: 1.5;
  color: var(--ternary);
  font-size: 1rem;
  font-weight: 400;
  box-sizing: border-box;
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 0.5rem 1.5rem;
  width: 100%;
}

.link {
  text-transform: uppercase;
  text-decoration: none;
  padding: 0.25rem 0.75rem;
  list-style: none;
}

.menu {
  width: 50%;
  display: flex;
  text-align: left;
}
