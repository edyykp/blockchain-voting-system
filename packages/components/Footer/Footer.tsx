import { Link } from 'react-scroll';

import { useMenu, useSiteProperties } from '@packages/config';

import styles from './Footer.module.css';

export const Footer = () => {
  const menu = useMenu();
  const valueOf = useSiteProperties();

  const text = {
    author: valueOf('site_author'),
  };

  return (
    <footer className={styles.footer} data-testid="footer">
      <ul className={styles.menu}>
        {menu.map((item, key) => (
          <Link to={item.url} className={styles.link} key={key}>
            {item.title}
          </Link>
        ))}
      </ul>
      <div className={styles.copyright}>
        Â© 2022, Proudly Coded by {text.author}
      </div>
    </footer>
  );
};
