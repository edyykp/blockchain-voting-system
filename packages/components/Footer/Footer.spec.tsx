import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

import { Footer } from './Footer';

const setup = () => {
  return {
    ...render(<Footer />),
  };
};

describe('Footerâ†’', () => {
  it('renders', () => {
    const { container, getByTestId } = setup();

    expect(getByTestId('footer')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
