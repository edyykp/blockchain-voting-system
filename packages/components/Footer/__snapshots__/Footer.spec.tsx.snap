// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer→ renders 1`] = `
<div>
  <footer
    class="footer"
    data-testid="footer"
  >
    <ul
      class="menu"
    >
      <a
        class="link"
      >
        About
      </a>
      <a
        class="link"
      >
        Data Protection
      </a>
    </ul>
    <div
      class="copyright"
    >
      © 2022, Proudly Coded by 
      Eduard-Constantin Stoica
    </div>
  </footer>
</div>
`;
